training/datastore/s3_connection.py

```diff
@@ -79,3 +88,159 @@ def load_existing_data(self):
+            raise Exception("S3 could not read the file.")
+
+    def create_package_train_user_data(self):
+        """Create package train user data."""
+        self.package_train_user_data = list()
+        for user_id in range(self.num_users):
+            this_user_items = self.pairs_train[self.pairs_train[:, 0] == user_id, 1]
+            items_str = " ".join(str(x) for x in this_user_items)
+            self.package_train_user_data.append([len(this_user_items), items_str])
+        return self.package_train_user_data
+
+    def create_package_train_item_data(self):
+        """Create package train item data."""
+        self.package_train_item_data = list()
+        for item_id in range(self.num_items):
+            this_item_users = self.pairs_train[self.pairs_train[:, 1] == item_id, 0]
+            users_str = " ".join(str(x) for x in this_item_users)
+            self.package_train_item_data.append([len(this_item_users), users_str])
+        return self.package_train_item_data
+
+    def create_package_test_user_data(self):
+        """Create package test user data."""
+        self.package_test_user_data = list()
+        for user_id in range(self.num_users):
+            this_user_items = self.pairs_test[self.pairs_test[:, 0] == user_id, 1]
+            items_str = " ".join(str(x) for x in this_user_items)
+            self.package_test_user_data.append([len(this_user_items), items_str])
+        return self.package_test_user_data
+
+    def create_package_test_item_data(self):
+        """Create package test item data."""
+        self.package_test_item_data = list()
+        for item_id in range(self.num_items):
+            this_item_users = self.pairs_test[self.pairs_test[:, 1] == item_id, 0]
+            users_str = " ".join(str(x) for x in this_item_users)
+            self.package_test_item_data.append([len(this_item_users), users_str])
+        return self.package_test_item_data
+
+    def train_test_data(self):
+        """Create the training testing data for PMF."""
+        data_list = self.split_training_testing_data()
+        self.pairs_train = data_list[0]
+        self.pairs_test = data_list[1]
+        self.num_users = data_list[2]
+        self.num_items = data_list[3]
+        user_train_data = self.create_package_train_user_data()
+        item_train_data = self.create_package_train_item_data()
+        user_test_data = self.create_package_test_user_data()
+        item_test_data = self.create_package_test_item_data()
+        return user_train_data, item_train_data, user_test_data, item_test_data
+
+    def split_training_testing_data(self):
+        """Split data into training and testing."""
+        data_in_bytes = self.load_user_item_data()
+        data = data_in_bytes.decode("utf-8")
+        data_list = data.split('\n')
+        pairs_train = []
+        pairs_test = []
+        user_id = 0
+        np.random.seed(123)
+        logger.info("Splitting data into training and testing.")
```

Comment:
It's good practice to seed with the time instead of the same number always. 
```python
    import numpy, time
    numpy.random.seed(time.time())
```

Reply:
Oh, a new thing. I will apply that, thanks !!

Reply:
Please take care of this as well, currently the same random number will be generated every time.